homeassistant/components/tfl/config_flow.py

```diff
@@ -0,0 +1,230 @@
+        error_code = exception.code
+        if error_code in (429, 401, 403):
+            raise InvalidAuth from exception
+
+        if error_code == 404:
+            raise ValueError from exception
+
+        raise
+    except URLError as exception:
+        raise CannotConnect from exception
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Transport for London."""
+
+    VERSION = 1
+    data: dict[str, Any] = {}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            try:
+                await validate_app_key(self.hass, user_input[CONF_API_APP_KEY])
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                # Input is valid, set data
+                self.data = user_input
+                self.data[CONF_STOP_POINTS] = []
+                return await self.async_step_stop_point()
+
+        return self.async_show_form(
+            step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_stop_point(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the stop point step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            try:
+                app_key = self.data[CONF_API_APP_KEY]
+                await validate_stop_point(
+                    self.hass,
+                    app_key,
+                    user_input[CONF_STOP_POINT],
+                )
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except ValueError:
+                errors["base"] = "invalid_stop_point"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                # User input is valid, save the stop point
+                self.data[CONF_STOP_POINTS].append(user_input[CONF_STOP_POINT])
+                # If user ticked the box show this form again so they can add an
+                # additional stop point.
+                if user_input.get(CONF_STOP_POINT_ADD_ANOTHER, False):
+                    return await self.async_step_stop_point()
+
+                # return await self.async_step_stop_points()
+                return self.async_create_entry(
+                    title="Transport for London", data=self.data
```

Comment:
Instead of storing all the user input in config entry data, and then copying it to user entry option in the options flow, store the settings which the user should be able to change in options already in the config flow.

Reply:
Ah, definitely felt I was missing something here. Docs and tutorials I'd seen tend to keep the data separate, which led me in the wrong direction.
Your suggestion is… obvious in hindsight. Thanks!

Reply:
Done